packaging/package.sh

```diff
@@ -18,26 +18,37 @@
 
 function usage() {
     echo ""
-    echo "usage: ./package.sh [-p|--pack] [-h|--help] [ARGS]"
+    echo "usage: ./package.sh [-h|--help] -d|--distribution <name> [-r|--release <version>] [-p|--pack oss|OSS|noredist|NOREDIST] [-s|--simulator default|DEFAULT|simulator|SIMULATOR]"
     echo ""
-    echo "The commonly used Arguments are:"
-    echo "-p|--pack oss|OSS             To package with only redistributable libraries (default)"
-    echo "-p|--pack noredist|NOREDIST   To package with non-redistributable libraries"
-    echo "-d centos7|centos63|fedora20|fedora21  To build a package for a distribution"
-    echo "-s simulator|SIMULATOR        To build for Simulator"
+    echo "The supported arguments are:"
+    echo "  To package with only redistributable libraries (default)"
+    echo "    -p|--pack oss|OSS"
+    echo "  To package with non-redistributable libraries"
+    echo "    -p|--pack noredist|NOREDIST"
+    echo "  To build a package for a distribution (mandatory)"
+    echo "    -d|--distribution centos7|centos63|fedora20|fedora21"
+    echo "  To set the package release version (optional)"
+    echo "  (default is 1 for normal and prereleases, empty for SNAPSHOT)"
+    echo "    -r|--release version(integer)"
+    echo "  To build for Simulator (optional)"
+    echo "    -s|--simulator default|DEFAULT|simulator|SIMULATOR"
+    echo "  To display this information"
+    echo "    -h|--help"
     echo ""
-    echo "Examples: ./package.sh -p|--pack oss|OSS"
-    echo "          ./package.sh -p|--pack noredist|NOREDIST"
-    echo "          ./package.sh (Default OSS)"
-    exit 1
+    echo "Examples: ./package.sh --pack oss"
+    echo "          ./package.sh --pack noredist"
+    echo "          ./package.sh --pack oss --distribution centos7 --release 42"
```

Comment:
Wilder,

This does not affect the contents in any way, just makes it easier to use with "yum/apt-get update" for e.g. when you want to issue updates.
It will come in handy as newer jenkins (as well as manual) built packages will now be able to gracefully upgrade old ones.
I used to fiddle around with cloud.spec to achieve the same and the idea was discussed last year with ke4qqq & spark404 but we never got anywhere. :)

Good job @davidamorimfaria !

--
Sent from the Delta quadrant using Borg technology!

Nux!
www.nux.ro

----- Original Message -----
> From: "Wilder Rodrigues" <notifications@github.com>
> To: "apache/cloudstack" <cloudstack@noreply.github.com>
> Cc: "NuxRo" <nux@li.nux.ro>
> Sent: Wednesday, 18 November, 2015 07:58:40
> Subject: Re: [cloudstack] CLOUDSTACK-9065: Packaging RPM, add option for package release version, cleanup and lint
> (#1075)

>>      echo ""
>> -    echo "Examples: ./package.sh -p|--pack oss|OSS"
>> -    echo "          ./package.sh -p|--pack noredist|NOREDIST"
>> -    echo "          ./package.sh (Default OSS)"
>> -    exit 1
>> +    echo "Examples: ./package.sh --pack oss"
>> +    echo "          ./package.sh --pack noredist"
>> +    echo "          ./package.sh --pack oss --distribution centos7 --release
>> 42"
> 
> @davidamorimfaria, what are the implications in giving a release number? For
> example, giving 4.6 or nothing will reflect in a different package or is it
> just for the final RPM file name?
> 
> For the looks on the code, it seems to be only the file name. :)
> 
> No big deal, I'm just curious. I will take some time to test your PR today.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/apache/cloudstack/pull/1075/files#r45169717